src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs

```diff
@@ -312,15 +317,7 @@ public ShapeResult OnGroup(params string[] groupIds)
     {
         ArgumentNullException.ThrowIfNull(groupIds);
 
-        foreach (var groupId in groupIds)
-        {
-            if (string.IsNullOrEmpty(groupId))
-            {
-                continue;
-            }
-
-            _groupIds = StringValues.Concat(_groupIds, groupId);
-        }
+        _groupIds = StringValues.Concat(_groupIds, groupIds.Where(x => x != null).ToArray());
```

Comment:
This code is allocating however

Reply:
that linq statement is not needed. fixed.